gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -391,6 +392,10 @@ private void pollAndAdvanceDag()
             jobExecutionPlan.setExecutionStatus(RUNNING);
             break;
         }
+
+        if (jobStatus.isShouldRetry()) {
```

Comment:
Only failed job's status will have this flag turned on?

Reply:
Yes, and they also have their status updated to running at the same time.